genai-perf/genai_perf/inputs/converters/template_converter.py

```diff
@@ -0,0 +1,111 @@
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import json
+import os
+from typing import Any, Dict, cast
+
+import jinja2
+from genai_perf.exceptions import GenAIPerfException
+from genai_perf.inputs.converters.base_converter import BaseConverter
+from genai_perf.inputs.inputs_config import InputsConfig
+from genai_perf.inputs.retrievers.generic_dataset import GenericDataset
+
+NAMED_TEMPLATES = {
+    "nv-embedqa": """[
+        {% for item in texts %}
+            {"text": [{{ item|tojson }}]}{% if not loop.last %},{% endif %}
+        {% endfor %}
+    ]""",
+}
+
+
+class TemplateConverter(BaseConverter):
+    """
+    Convert a GenericDataset to a request body.
+
+    The template should render a list of text strings, and return a list of payloads.
+    """
+
+    def resolve_template(self, template_name_or_content: str):
+
+        if template_name_or_content in NAMED_TEMPLATES:
+            environment = jinja2.Environment(
+                autoescape=True,
+                loader=jinja2.DictLoader(NAMED_TEMPLATES),
+            )
+            return environment.get_template(template_name_or_content)
+
+        if os.path.isfile(template_name_or_content):
+            try:
+                with open(template_name_or_content, "r", encoding="utf-8") as f:
+                    template_content = f.read()
+            except Exception as e:
+                raise GenAIPerfException(f"Error reading template file: {e}")
+        else:
+            raise GenAIPerfException(
+                f"Template file not found: {template_name_or_content}"
+            )
+
+        environment = jinja2.Environment(autoescape=True)
+        return environment.from_string(template_content)
+
+    def check_config(self, config: InputsConfig) -> None:
+        output_template = config.extra_inputs.get("output_template")
+        if not output_template:
+            raise GenAIPerfException(
+                f"The template converter requires the "
+                "extra input output_template, only "
+                "detected the following --extra-inputs: "
+                "{config.extra_inputs.keys}."
+            )
+        try:
+            template = self.resolve_template(output_template)
+            test_texts = ["test1", "test2"]
+            payloads_json = template.render(texts=test_texts)
+            payloads = json.loads(payloads_json)
+            if not isinstance(payloads, list):
+                raise ValueError(
+                    "Template does not render a list of strings to a list of items. "
+                    f"For example, {test_texts} is rendered into {payloads}."
+                )
+        except Exception as e:
+            raise GenAIPerfException(e)
+
+    def convert(
+        self, generic_dataset: GenericDataset, config: InputsConfig
+    ) -> Dict[Any, Any]:
+        output_template = config.extra_inputs.get("output_template")
```

Comment:
same as above, why output template? I think output of the server when i hear output template, and think it might be confusing.

Reply:
This is a good catch. I was using the original PR terminology. I think "payload template" might work. So that it's clear what the template is for and it won't class with other templates we use in the future.